amethyst_input/src/controller.rs

```diff
@@ -0,0 +1,74 @@
+use event::InputEvent;
+
+/// Controller axes matching SDL controller model
+#[derive(Eq, PartialEq, Debug, Copy, Clone, Serialize, Deserialize)]
+pub enum ControllerAxis {
+    LeftX,
+    LeftY,
+    RightX,
+    RightY,
+    LeftTrigger,
+    RightTrigger,
+}
+
+/// Controller buttons matching SDL controller model
+#[derive(Eq, PartialEq, Debug, Copy, Clone, Serialize, Deserialize)]
+pub enum ControllerButton {
+    A,
+    B,
+    X,
+    Y,
+    DPadDown,
+    DPadLeft,
+    DPadRight,
+    DPadUp,
+    LeftShoulder,
+    RightShoulder,
+    LeftStick,
+    RightStick,
+    Back,
+    Start,
+    Guide,
+}
+
+#[derive(PartialEq, Debug, Copy, Clone, Serialize, Deserialize)]
+pub enum ControllerEvent {
+    ControllerAxisMoved {
+        which: u32,
```

Comment:
(Optionnal) Rename which to controller_id, to stay in line with how we name things in amethyst usually.

Reply:
I originally had it as `controller_id`, but i changed it to `which` to signal that this is something different than `controller_id` seen from the user code. There is a mapping between those two. If anything, `controller_index` could be a better name, but it still stays confusing.

Reply:
kk